### ISSUE TITLE
Using middleware with 1.x clients

### ISSUE THREAD

**Issue by memoselyk** [kubernetes-client/javascript#1925]
I gather from the docs, that the standard pattern to create a client is with `KubeConfig`'s `makeApiClient` (see [here](https://github.com/kubernetes-client/javascript/blob/release-1.x/src/config.ts#L459) ).

However, there is no way to create a client with middlewares.

My use case is to add a proxy or use impersonation headers to a client, that was possible in the client's instance with `apiclient.addInterceptor` and it seem the new mechanism is using middlewares.

What would be the best place to give a middleware?
Should it be an argument to the `makeApiClient` _factory_ or is it better to make them part of the kube config, eg, `config.addMiddlewares`?

I feel inclined towards the first option, but I'm relatively new to this library.

**Comment by Noname-OmerR** [comment]
Ran into the same issue.
Previously I used v.022, but it has security vulnerabilities.

How can you add a proxy configuration to the new v1.x clients?

**Comment by brendandburns** [comment]
Yeah, I think that's fine, as long as it is optional, we can't break existing callers of `makeApiClient`
